### PR TITLE
BB-4472: Add support for fixed tax rates in ecommerce (WIP)

### PR DESCRIPTION
**Pull request by giovannicimolin**
This introduces support for a fixed tax rate setting in e-commerce through two new django settings (`ENABLE_TAX_RATE` and
`TAX_RATE`).
Default values preserve the previous platform behavior.

**Testing instructions:**
TBD

**Reviewers:**
- [ ] @dliberat 
- [ ] @lgp171188

### CODE REVIEW

File: ecommerce/extensions/partner/strategy.py [hunk 1]
```diff
@@ -33,8 +36,25 @@ def availability_policy(self, product, stockrecord):
...
+        Only return tax rate if it's enabled in settings, otherwise return 0.
+        """
+        if not settings.ENABLE_TAX_RATE:
+            return D('0')
+        return self.rate
+
+
 class DefaultStrategy(strategy.UseFirstStockRecord, CourseSeatAvailabilityPolicyMixin,
-                      strategy.NoTax, strategy.Structured):
+                      TaxStrategyMixin, strategy.Structured):
```
Comment (lgp171188) [inline ecommerce/extensions/partner/strategy.py:57]: @giovannicimolin, wouldn't it be possible to define the mixin and a new strategy class in PayTabs plugin app and use it as the strategy for PayTabs, without requiring this code drift?

File: ecommerce/settings/base.py [hunk 2]
```diff
@@ -834,3 +834,7 @@
 # To check government purchase restriction lists
 SDN_CHECK_API_URL ="https://api.trade.gov/gateway/v1/consolidated_screening_list/search"
 SDN_CHECK_API_KEY = "sdn search key here"
+
+# Enable and disable tax rates
+ENABLE_TAX_RATE = True
+TAX_RATE = '0.3'
```
Comment (lgp171188) [inline ecommerce/settings/base.py:840]: @giovannicimolin, these can and should be defined via Ansible configuration (`ECOMMERCE_EXTRA_CONFIG_OVERRIDES`).

File: ecommerce/extensions/partner/strategy.py (hunk 1 above)
Comment (giovannicimolin) [reply ecommerce/extensions/partner/strategy.py:57]: @lgp171188 No, in that case, the value including tax would not be displayed to the user in the basket (before anything related to the payment provider happens).
This is one of our requirements, so I'm implementing this to give proper tax support throughout the application.

File: ecommerce/settings/base.py (hunk 2 above)
Comment (giovannicimolin) [reply ecommerce/settings/base.py:840]: @lgp171188 There's no need for defining variables there, we can leave sane defaults here (`TAX_RATE = '0.0'` and maybe remove the `ENABLE_TAX_RATE` variable) and use `ECOMMERCE_EXTRA_CONFIG_OVERRIDES` to change this during configuration.

Values on `ECOMMERCE_EXTRA_CONFIG_OVERRIDES` will automatically override any configuration variable as long as they are defined in `ecommerce.yml`.

**Pull request closed without merging**